DataPerProcessing/utilities.py

```diff
@@ -90,27 +124,81 @@ def polygonToRotRectangle(bbox):
     bbox = np.array(bbox,dtype=np.float32)
     bbox = np.reshape(bbox,newshape=(2,4),order='F')
     angle = math.atan2(-(bbox[0,1]-bbox[0,0]),bbox[1,1]-bbox[1,0])
-
     center = [[0],[0]]
-
     for i in range(4):
         center[0] += bbox[0,i]
         center[1] += bbox[1,i]
 
     center = np.array(center,dtype=np.float32)/4.0
-
     R = np.array([[math.cos(angle), -math.sin(angle)], [math.sin(angle), math.cos(angle)]], dtype=np.float32)
-
     normalized = np.matmul(R.transpose(),bbox-center)
-
     xmin = np.min(normalized[0,:])
     xmax = np.max(normalized[0,:])
     ymin = np.min(normalized[1,:])
     ymax = np.max(normalized[1,:])
-
     w = xmax - xmin + 1
     h = ymax - ymin + 1
-
     return [float(center[0]),float(center[1]),w,h,angle]
 
-#print(polygonToRotRectangle(getBBoxList(labelPath+os.sep+'P0000.txt')[0]))
\ No newline at end of file
+#print(polygonToRotRectangle(getBBoxList(labelPath+os.sep+'P0000.txt')[0]))
+
+#------------------------------------------------------------------------
+# Function Name: cropImageBBox()
+# Description: Crop the object and make bg black
+#------------------------------------------------------------------------
+def cropImageBBox(imagePath, bbox, mode='show', outPath=None):
+	'''
+	* imagePath: raw image path from data set
+	* bbox     : The polygon stored in format [x1, y1, x2, y2, x3, y3, x4, y4]
+	* mode     : 'show' or 'save'
+	* outPath  : if mode is 'save' then output path
+	'''
+	x1, y1, x2, y2, x3, y3, x4, y4 = bbox
+	image = cv2.imread(imagePath)
+	mask = np.zeros(image.shape, dtype=np.uint8)
+	# mask defaulting to black for 3-channel and transparent for 4-channel
+	# (of course replace corners with yours)
+	roi_corners = np.array([[(x1,y1), (x2,y2), (x3,y3), (x4,y4)]], dtype=np.int32)
+	# fill the ROI so it doesn't get wiped out when the mask is applied
+	channel_count = image.shape[2]  # i.e. 3 or 4 depending on your image
+	ignore_mask_color = (255,)*channel_count
+	cv2.fillPoly(mask, roi_corners, ignore_mask_color)
+	# from Masterfool: use cv2.fillConvexPoly if you know it's convex
+
+	# apply the mask
+	masked_image = cv2.bitwise_and(image, mask)
+	xmin = min(x1,x2,x3,x4)
+	xmax = max(x1,x2,x3,x4)
+	ymin = min(y1,y2,y3,y4)
+	ymax = max(y1,y2,y3,y4)
+	image = masked_image[ymin:ymax, xmin:xmax] 
+	# save the result
+	if mode == 'show':
+		cv2.imshow('Image',image)
+		cv2.waitKey()
+	elif mode == 'save':
+		if outPath is not None:
+			cv2.imwrite(outPath, image)
+		else:
+			print('No Output path Provided')
+
+#------------------------------------------------------------------------
+# Function Name: sendToCropImageBBox()
+# Description: Preprocessing before sending to cropImageBBox
+#------------------------------------------------------------------------
+def sendToCropImageBBox(mainPath):
+	'''
+	* mainPath: Path where sub-folder with object class name will create
+	'''
+	ObjectCoordinatesList, ObjectNamesList, ObjectDifficultyList = getBBoxList(labelPath+os.sep+labelName)
```

Comment:
instead of making three separate list, a json containing three key({objcoord:coord, objname:name, objdiff: diff) will be much better and efficient to use, so final ouput which getBBoxlist should return is list of dict.